assets/js/modules/analytics/util/report-date-range-args.test.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * Tests for reporting date range utilities.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { generateDateRangeArgs } from './report-date-range-args';
+
+describe( 'Analytics reporting date range arguments', () => {
+	describe( 'generateDateRangeArgs', () => {
+		const allArgs = {
+			startDate: 'start-date',
+			endDate: 'end-date',
+			compareStartDate: 'compare-start-date',
+			compareEndDate: 'compare-end-date',
+		};
+
+		describe( 'if `startDate` or `endDate` arguments are not provided', () => {
+			it( 'should return args untouched', () => {
+				const { startDate, endDate } = allArgs;
+
+				const noArgsPassed = generateDateRangeArgs();
+				const startDatePassed = generateDateRangeArgs( { startDate } );
+				const endDatePassed = generateDateRangeArgs( { endDate } );
+
+				expect( noArgsPassed ).toBe( undefined );
+				expect( startDatePassed ).toMatchObject( { startDate } );
+				expect( endDatePassed ).toMatchObject( { endDate } );
+			} );
```

Comment:
As in the Search Console PR, let's not return these arguments when they aren't valid. Whichever method you use in #2634 let's do here as well. 🙂